### ISSUE TITLE
Migrate AdSense Summary dashboard widget to use Widgets API

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#1901]
The current AdSense `AdSenseDashboardMainSummary` widget should be migrated to use the Widgets API (which is behind a feature flag), while maintaining the current version as legacy components. Those will be removed only once all widgets have been migrated so that we can do the entire switch.

Some general practices for migrating:
* The migration should happen by fully **re-implementing** the components: In other words, the existing components should keep their current functionality, they should **not** use the datastore. Those legacy components should be prefixed with `Legacy*`.
* A good starting point is probably to copy the components into new files and go from there, while renaming the old ones and leaving them untouched.
* For the refactored version, _only_ the actual widget components should be fully responsible for retrieving data from the datastore. If they rely on further `common` components for UI, these should receive such data via props.
* If there are UI parts that can be reused between the new and legacy components, those can be outsourced into `common` components, however as mentioned above **it is important that the legacy components keep using the legacy data API and are not switched to the datastore**. Only the new ones should be.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The [`AdSenseDashboardMainSummary`](https://github.com/google/site-kit-wp/blob/1.13.1/assets/js/modules/adsense/components/dashboard/AdSenseDashboardMainSummary.js) component should be migrated to a functional component using the datastore entirely, no globals and no legacy data API.
* Refactoring map:
    * `dashboard/AdSenseDashboardMainSummary.js` should become `dashboard/DashboardSummaryWidget.js` (refactored) and `dashboard/LegacyAdSenseDashboardMainSummary.js` (legacy). The refactored component should access the datastore for the main API data.
    * `dashboard/DashboardEarnings.js` should simply be renamed to `dashboard/LegacyDashboardEarnings.js`.
* The new `DashboardSummaryWidget` should return `null` if the AdSense module is not active. _Only_ otherwise AdSense data should be requested, i.e. this probably needs to go into a separate component `DashboardSummary`, since hooks cannot run conditionally (we use `*Widget` as top-level entry point while the un-suffixed version is essentially the main part of it, without any extra checks for active modules etc.).
* The new `DashboardSummaryWidget` component should be registered as a widget with priority 1, width `Widgets.WIDGET_WIDTHS.HALF`, and for the widget area `AREA_DASHBOARD_EARNINGS`.

#### Stage 1
* Rename the following files and components accordingly:
  * `assets/js/modules/adsence/dashboard/AdSenseDashboardMainSummary.js` -> `assets/js/modules/adsence/dashboard/LegacyAdSenseDashboardMainSummary.js`;
  * `assets/js/modules/adsence/dashboard/DashboardEarnings.js` -> `assets/js/modules/adsence/dashboard/LegacyDashboardEarnings.js`.
* Update `index.legacy.js` file to use `LegacyDashboardEarnings` instead of `DashboardEarnings`.

#### Stage 2
* Create `assets/js/modules/adsence/dashboard/DashboardSummaryWidget.js` file;
* `DashboardSummaryWidget.js` file should:
  * Define a new functional component with `DashboardSummaryWidget` name;
  * Use `useSelect` hook and the following selectors:
    * `getDateRange` selector from the `core/user` datastore to retrieve the current date range;
    * `getReport` selector from the adsense datastore (implemented in #1776):
      * to retrieve earnings information for selected date range;
      * to retrieve earnings information for today (use `startDate` and `endDate`);
      * to retrieve earnings information for the last month (use `startDate` and `endDate`) with `'DATE'` dimension.
  * Replicate the logic from the legacy comonent to render summary widget;
  * Wrap rendered component with the `Widget` component that has `adsenseSummary` slug  and `googlesitekit-widget--no-padding` class, and return it.
* Export a component enhanced with `whenActive` HOC (see Stage 2 in #1896) that uses `adsence` as module name.

#### Stage 3
* Update `assets/js/modules/adsence/index.js` file:
  * Import `domReady` from `@wordpress/dom-ready` and `Widgets` from `googlesitekit-widgets`;
  * Call `domReady` with a callback function that registers a new widget using `Widgets.registerWidget` function and passing the following arguments:
    * A string `adsenseSummary` as widget slug;
    * Settings object where:
      * `component` is `DashboardSummaryWidget`;
      * `width` is `Widgets.WIDGET_WIDTHS.HALF`;
      * `priority` is `1`;
      * `wrapWidget` is `false`.
    * Areas array that includes just `AREA_DASHBOARD_EARNINGS` imported from `googlesitekit/widgets/default-areas`.

## QA Brief

* In a development build, the new AdSense Summary widget hould render correctly in the Site Kit dashboard. It should behave the same way as the old ones.
* It should be tested that they display errors if e.g. the API request goes wrong - they shouldn't keep loading infinitely in such a case (try e.g. to block client-to-server requests to cause an error).
* The old/legacy AdSense Summary widget should continue to show up in both development and production builds and behave like before.

## Changelog entry

* Migrate AdSense Summary widget to new Widget API.

**Comment by felixarntz** [comment]
IB ✅

**Comment by eugene-manuilov** [comment]
@felixarntz since metrics are required now for AdSense `getReport` selector, which ones do we need to use for `DashboardSummaryWidget` widget?

**Comment by felixarntz** [comment]
@eugene-manuilov We'll need to use the ones needed for the three pieces of data in that widget, which are the ones that used to be automatically handled on the server (and still are now for backward-compatibility): `'EARNINGS', 'PAGE_VIEWS_RPM', 'IMPRESSIONS'`

**Comment by cole10up** [comment]
**Tested**

Installed latest develop, setup and activated Adsense

Noticed on the dashboard:
![image](https://user-images.githubusercontent.com/57115907/94110278-3d92c600-fdf7-11ea-85c1-1d4369abc6c8.png)

Compared to legacy widget:
![image](https://user-images.githubusercontent.com/57115907/94110313-4a171e80-fdf7-11ea-9e0c-3ca42e22c63f.png)

Sending back to address the findings @felixarntz - this is on hellogoodbye.today

**Comment by aaemnnosttv** [comment]
Thanks @cole10up! Taking a closer look at this it seems due to the component selecting the `error` for the `getReport` selector. I think we just need to update the logic to ignore the error in this case since it is expected.

**Comment by aaemnnosttv** [comment]
@felixarntz I just realized that the component I fixed is unrelated to this issue and PR 😄 . It's still good to fix and @cole10up was correct to point that out, but probably shouldn't block this issue from moving forward (assuming everything else was good).

I'm going to create a new issue for the problem Cole found here and assign this back to @cole10up for a final pass in QA.

**Comment by cole10up** [comment]
**Final pass looks good**

![image](https://user-images.githubusercontent.com/57115907/94168613-ac494100-fe42-11ea-9c31-2f5df30f6d78.png)

Checked Mobile:
![image](https://user-images.githubusercontent.com/57115907/94168742-cc790000-fe42-11ea-85f4-4cfa4b6865e2.png)

Passed QA ✅
